src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java

```diff
@@ -908,6 +1036,124 @@ public void setNumOfRubricSubQuestions(int numOfRubricSubQuestions) {
+        int[][] numOfResponsesPerSubQuestionPerChoice;
+        double[] totalPerSubQuestion;
+
+        RubricRecipientStatistics(String recipient, String recipientName, String recipientTeam) {
+            this.recipientEmail = recipient;
+            this.recipientName = recipientName;
+            this.recipientTeam = recipientTeam;
+            numOfResponsesPerSubQuestionPerChoice = new int[getNumOfRubricSubQuestions()][getNumOfRubricChoices()];
+            totalPerSubQuestion = new double[getNumOfRubricSubQuestions()];
+        }
+
+        public void addResponseToRecipientStats(FeedbackResponseAttributes response) {
+            if (!response.recipient.equalsIgnoreCase(recipientEmail)) {
+                return;
+            }
+
+            FeedbackRubricResponseDetails rubricResponse = (FeedbackRubricResponseDetails) response.getResponseDetails();
+
+            for (int i = 0; i < getNumOfRubricSubQuestions(); i++) {
+                int choice = rubricResponse.getAnswer(i);
+
+                if (choice >= 0) {
+                    ++numOfResponsesPerSubQuestionPerChoice[i][choice];
+                    totalPerSubQuestion[i] += getRubricWeights().get(choice);
+                }
+            }
+        }
+
+        public String getHtmlForSubQuestion(int subQuestion) {
+            StringBuilder html = new StringBuilder(100);
+            String alphabeticalIndex = StringHelper.integerToLowerCaseAlphabeticalIndex(subQuestion + 1);
+            String subQuestionString = SanitizationHelper.sanitizeForHtml(alphabeticalIndex + ") "
+                    + getRubricSubQuestions().get(subQuestion));
+            DecimalFormat df = new DecimalFormat("#");
+            DecimalFormat dfAverage = new DecimalFormat("0.00");
+
+            List<String> cols = new ArrayList<>();
+
+            // <td> entries which display recipient identification details and rubric subQuestion
+            cols.add(recipientTeam);
+            cols.add(recipientName);
+            cols.add(subQuestionString);
+
+            // <td> entries which display number of responses per subQuestion per rubric choice
+            for (int i = 0; i < getNumOfRubricChoices(); i++) {
+                cols.add(Integer.toString(numOfResponsesPerSubQuestionPerChoice[subQuestion][i]));
+            }
+
+            // <td> entries which display aggregate statistics
+            cols.add(df.format(totalPerSubQuestion[subQuestion]));
+            cols.add(dfAverage.format(totalPerSubQuestion[subQuestion] / getNumOfRubricSubQuestions()));
+
+            // Generate HTML for all <td> entries using template
+            for (String col : cols) {
+                html.append(
+                        Templates.populateTemplate(FormTemplates.RUBRIC_RESULT_RECIPIENT_STATS_BODY_ROW_FRAGMENT,
+                        Slots.RUBRIC_RECIPIENT_STAT_CELL, col));
+            }
+
+            return html.toString();
+        }
+
+        public String getHtmlForAllSubQuestions() {
```

Comment:
Likewise, document that this returns a sequence of `<tr>...</tr><tr>...</tr>...` without an enclosing `<tbody>...</tbody>`.